auraed/src/runtime/cell_service/cells/cgroups/max.rs

```diff
@@ -0,0 +1,75 @@
+ *             Apache 2.0 License Copyright © 2022 The Aurae Authors          *
+ *                                                                            *
+ *                +--------------------------------------------+              *
+ *                |   █████╗ ██╗   ██╗██████╗  █████╗ ███████╗ |              *
+ *                |  ██╔══██╗██║   ██║██╔══██╗██╔══██╗██╔════╝ |              *
+ *                |  ███████║██║   ██║██████╔╝███████║█████╗   |              *
+ *                |  ██╔══██║██║   ██║██╔══██╗██╔══██║██╔══╝   |              *
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+use std::fmt::{Display, Formatter};
+use std::ops::Deref;
+use validation::{ValidatedField, ValidationError};
+
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub struct Max(i64);
+
+impl Max {
+    #[cfg(test)]
+    pub fn new(max: i64) -> Self {
+        Self(max)
+    }
+
+    pub fn into_inner(self) -> i64 {
+        self.0
+    }
+}
+
+impl ValidatedField<i64> for Max {
+    fn validate(
+        input: Option<i64>,
+        field_name: &str,
+        parent_name: Option<&str>,
+    ) -> Result<Self, ValidationError> {
+        let input = validation::required(input, field_name, parent_name)?;
+
+        validation::minimum_value(input, 0, "units", field_name, parent_name)?;
```

Comment:
maybe i've misread this but shouldn't this _also_ have a validation for maximum_value 10000?

Reply:
There doesn't seem to be a requirement for that:

>Limits are in the range [0, max] and defaults to “max”, which is noop.
>
>As limits can be over-committed, all configuration combinations are valid and there is no reason to reject configuration changes or process migrations.

But, it doesn't seem to make sense to go over the maximum allowed weight (10000).

Reply:
that's what i was thinking. but maybe there are other "max"es that are not 10k.

Reply:
Hmm, reading the [docs](https://docs.kernel.org/admin-guide/cgroup-v2.html#resource-distribution-models) again. "max" is used in multiple contexts. Here we are referring to limits, so I think we should leave the validation, but rename the field/struct to limit